core/src/main/java/club/mondaylunch/gatos/core/codec/NodeConnectorCodecProvider.java

```diff
@@ -0,0 +1,22 @@
+package club.mondaylunch.gatos.core.codec;
+
+import org.bson.codecs.Codec;
+import org.bson.codecs.configuration.CodecProvider;
+import org.bson.codecs.configuration.CodecRegistry;
+
+import club.mondaylunch.gatos.core.graph.connector.NodeConnector;
+
+public enum NodeConnectorCodecProvider implements CodecProvider {
+
+    INSTANCE;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <T> Codec<T> get(Class<T> clazz, CodecRegistry registry) {
+        if (NodeConnector.class.isAssignableFrom(clazz)) {
+            return (Codec<T>) new NodeConnectorCodec(registry, clazz);
+        } else {
+            return null;
```

Comment:
What happens when a CodecProvider returns null? Does the thing just not get serialised, or is an exception thrown?

Reply:
I think the serializer goes through all the registered codec providers and uses the first one that doesn't return null. If all the codec providers returned null, an exception is thrown.